server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/symbols/SymbolsRepository.java

```diff
@@ -66,10 +67,13 @@ public void updateSymbols(Map<String, SymbolTable> symbols) {
    * Get variable data
    *
    * @param program the program node
-   * @return map of variables
+   * @return Collection of variables nodes
    */
   public Multimap<String, VariableNode> getVariables(ProgramNode program) {
-    return getSymbolTable(program).getVariables();
+    Multimap<String, VariableNode> result = ArrayListMultimap.create();
+    result.putAll(getSymbolTable(program).getVariablesMap());
+    result.putAll(getSymbolTable(program).getVariablesGlobalsMap());
```

Comment:
If this class is really a singleton accessed from multiple threads then these might be two different SymbolTables.